src/System.Text.Utf8/tests/RandomTests.cs

```diff
@@ -704,5 +704,31 @@ public void IndexOfUnicodeCodePoint(int expected, string s, uint codePointValue)
             UnicodeCodePoint codePoint = (UnicodeCodePoint)codePointValue;
             Assert.Equal(expected, u8s.IndexOf(codePoint));
         }
+
+        [Fact]
+        public void ReferenceEquals()
+        {
+            Utf8String s1 = new Utf8String("asd");
+            Assert.True(s1.ReferenceEquals(s1));
+
+            Utf8String s2 = new Utf8String("asd");
+            Assert.True(s2.ReferenceEquals(s2));
+
+            Assert.False(s1.ReferenceEquals(s2));
+            Assert.False(s2.ReferenceEquals(s1));
+        }
+
+        [Fact]
+        public void ConstructingFromUtf16StringWithNullValueThrowsArgumentException()
+        {
+            Assert.Throws<ArgumentException>(() => { Utf8String utf8String = new Utf8String((string)null); });
+        }
+
+        [Fact]
+        public void ConstructingFromUtf16StringWithEmptyStringDoesNotAllocate()
```

Comment:
This test does not prove an allocation did not take place, only that various mechanisms of creating an empty string return Utf8Strings that point to the same buffer. Perhaps *EmptyUtf8StringsShareASingletonBuffer*?